### PR TITLE
Resolve resumed claude through the cmux wrapper shim env var (fixes #5639)

### LINKED ISSUE #5639: Claude resume still drops cmux hooks after #5430: bare `claude` doesn't resolve to the wrapper inside the $SHELL -lic restore launcher
## Summary

#5430 (fixing #5427) does **not** actually fix hook loss on resumed Claude sessions. Hooks (SessionStart/Stop/Notification/status pills) still fail to fire after closing and reopening cmux. The fix corrected the resume **argv string** but not the **shell context that executes it**, so the wrapper it depends on is never invoked.

Verified by reading the shipped code on `main` and running the wrapper directly.

## What #5430 changed

`AgentResumeArgv.claudeResumeArgv` now emits a **bare** `claude`:

```swift
return ["claude", "--resume", sessionId] + preserved
```

It intentionally does **not** add `--settings` itself — it relies on cmux's `claude` wrapper (`Resources/bin/cmux-claude-wrapper`) re-injecting the hook `--settings` when it sees `--resume`. That part is correct: running the wrapper as `claude --resume <id>` does produce `--settings <HOOKS_JSON> --resume <id>`. ✅

The fix's correctness rests entirely on the comment in `claudeResumeArgv`:
> "the `claude` wrapper … is first on `PATH` inside cmux terminals."

That assumption does not hold in the code path that auto-resume actually uses.

## Root cause

`claude` only resolves to the wrapper because the **cmux shell integration** (`Resources/shell-integration/cmux-*-integration.*`) does two things when sourced (`cmux-bash-integration.bash:206-275`):

1. installs the per-surface PATH shim `$TMPDIR/cmux-cli-shims/<surface>/claude` and prepends it to `PATH`, and
2. defines a `claude()` shell function that execs the wrapper.

Neither lives in the user's own `.zshrc`/`.zprofile`; they exist only while cmux's integration is active in that shell.

The local close-&-reopen restore uses `resumeStartupCommand()` (`Workspace.swift:1687`):

```swift
} else {
    restorableAgent?.resumeStartupCommand()        // local restore path
        .map(SurfaceResumeStartupLaunch.command)
}
```

which runs the agent through this launcher (`SessionPersistence.swift:1283`, `commandThenReturnLines`):

```zsh
_cmux_resume_shell="${SHELL:-/bin/zsh}"
case "${_cmux_resume_shell:t}" in
  zsh|bash) "$_cmux_resume_shell" -lic 'claude --resume <id>' ;;   # agent relaunched HERE
  ...
esac
# cmux integration ZDOTDIR is re-applied only AFTER, for the return shell:
export ZDOTDIR="$CMUX_SHELL_INTEGRATION_DIR"
...
exec -l "$_cmux_resume_shell"
```

The resumed agent is launched in a **fresh `$SHELL -lic` login-interactive shell**, and cmux's integration `ZDOTDIR` is restored **only for the post-exit `exec -l` return shell — not for the `-lic` invocation that relaunches the agent.** (The fact that the script re-exports `ZDOTDIR` at all confirms the team knows the user's login profile clobbers it.)

So inside that `-lic` shell the cmux `claude()` function and PATH shim are not in effect, bare `claude` resolves to the user's **real** binary (homebrew/npm/`~/.local/bin`), the wrapper is bypassed, no `--settings` is injected, and hooks stay dead — the exact original symptom.

This is also why **fresh launches work but resumes don't**: a fresh agent runs in the cmux terminal's integrated primary shell (function + shim present); restore relaunches it one level down in a login subshell where the integration isn't active.

## Deterministic repro

Inside any cmux terminal:

```sh
type claude                       # → cmux claude() wrapper function (good)
zsh -lic 'command -v claude'      # → /opt/homebrew/bin/claude (real binary, bad)
```

The second form is precisely the context the restore launcher uses. Then: start a Claude session, fully quit and reopen cmux, let it auto-resume — SessionStart/Stop/Notification hooks do not fire (no status pill, no notifications).

## Why this slipped through review

`ClaudeResumeHookSettingsTests` and the review bots only asserted the **argv string** (`["claude","--resume",…]`, stale settings stripped). Nothing exercised the **actual shell resolution** — i.e. that `claude` resolves to the wrapper inside the `$SHELL -lic` launcher. The fix was verified at the wrong layer.

## Fix directions

1. **Don't depend on PATH in the login subshell** — have the resume builder emit the wrapper's real path (e.g. `$CMUX_CLAUDE_WRAPPER_SHIM`, or the bundle's `Resources/bin/cmux-claude-wrapper`) instead of bare `claude`. Smallest, most robust.
2. Apply the cmux integration `ZDOTDIR`/PATH (shim) to the `-lic` command shell, not just the post-exit return shell.
3. Deliver the resume command as inline input to the already-integrated terminal shell (`resumeStartupInput`) instead of via the `$SHELL -lic` launcher.

A regression test should assert that the *executed* resume command runs through the wrapper (e.g. the launched argv contains the hook `--settings`), not just the pre-exec argv string.

Refs: #5427, #5430.

### PR DESCRIPTION
**Pull request by austinywang**
## Problem

#5430 (fixing #5427) corrected the resume **argv string** (bare `claude`) but not the **shell context that executes it**, so the cmux `claude` wrapper it depends on is still bypassed and resumed Claude sessions silently lose every cmux hook (SessionStart/Stop/Notification/status pills).

The close-&-reopen restore launcher runs the resumed agent via `$SHELL -lic <cmd>` (`SessionPersistence.commandThenReturnLines`), a fresh login-interactive shell where cmux's shell integration (the per-surface PATH shim + `claude()` function) is **not** active, and the rendered command is `env … claude …`. Two independent failures stack:

- `env` resolves `claude` via `execvp`, which **bypasses shell functions** entirely — so re-sourcing the integration would not help an `env`-prefixed invocation.
- the user's login profile rebuilds `PATH`, **dropping the integration's shim dir**.

So bare `claude` resolves to the user's real binary, no hook `--settings` is injected, and hooks stay dead — the exact original symptom. Fresh launches work because they run in the cmux terminal's integrated primary shell.

### Reproduced (deterministic, environment-independent)

Running the real `resumeCommand` through `zsh -lic` (what the launcher does) under a login profile that clobbers `PATH` and shadows `claude` with a function:

- **bare `claude`** (current `main`): `real (no hooks): --resume SID …` ❌
- **`"${CMUX_CLAUDE_WRAPPER_SHIM:-claude}"`** (this PR): `wrapper: --settings HOOKS_JSON --resume SID …` ✅

## Fix

Render the claude executable as the wrapper shim token `"${CMUX_CLAUDE_WRAPPER_SHIM:-claude}"` instead of bare `claude`. `CMUX_CLAUDE_WRAPPER_SHIM` is a **managed terminal env var** (`TerminalStartupEnvironment` / `GhosttyTerminalView`) inherited by every descendant shell **regardless of `PATH`/function shadowing**, so the executed resume/fork command execs the per-surface shim → `cmux-claude-wrapper` → re-injected hook `--settings` (the wrapper already re-adds `--settings` on `--resume`). It falls back to bare `claude` outside cmux (no regression). This is immune to the `env`-bypass / `PATH`-clobber that defeat integration-sourcing approaches (fix directions 2/3 in the issue); it is the issue's "smallest, most robust" direction 1.

Applied at **one shared value-layer seam** (`AgentResumeArgv.renderingClaudeWrapperExecutable`) across all three claude resume render paths, per the shared-behavior policy:
- app resume/fork builder (`AgentResumeCommandBuilder.shellCommand`)
- cmux-cli surface-restore publisher (`agentSurfaceResumeCommand`)
- session-index manual resume (`SessionEntry`)

**Class check:** other agents resumed through the same launcher (codex, etc.) replay an absolute executable path and never depended on a wrapper, so they are unaffected. `grok` relies on a function-only wrapper with no shim env var; it would need an analogous shim and is out of scope here.

## Tests (two-commit red/green)

1. **Failing test first** — `testClaudeResumeCommandExecutesThroughWrapperInsideLoginShellLauncher` runs the real `resumeCommand` through `zsh -lic` in a hermetic sandbox whose login profile clobbers `PATH` and shadows `claude` with a function, then asserts the launched invocation routed through the wrapper (its injected `--settings`). This asserts the **executed** command's shell resolution — the layer #5430's argv-string tests missed — and is red on `main`.
2. **The fix** — makes it green; existing `'claude' '--resume'` string expectations updated to the new token.

Package argv-layer tests are unchanged (the argv still emits a logical `claude`; the wrapper is reached only at render time), and pass.

## Localization

No user-facing strings added or changed — the token is an internal shell command fragment. No `Localizable.xcstrings` / web message updates required.

Fixes #5639. Refs #5427, #5430.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/5721"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783635369&installation_id=133855650&pr_number=5721&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F5721&signature=5aecaf791a906a93189c22656bebae1a4198a49f259e8ba8bf3062602c213ae9"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

---
## Summary by cubic
Routes resumed Claude sessions through the cmux wrapper using the `CMUX_CLAUDE_WRAPPER_SHIM` token, wrapped in `/bin/sh -c` for non‑POSIX shells. Restores hooks when reopening via `$SHELL -lic`, and safely falls back to PATH `claude` if the shim is missing or stale. Fixes #5639.

- **Bug Fixes**
  - Replace bare `claude` with an executable‑guarded wrapper shim token resolved from `CMUX_CLAUDE_WRAPPER_SHIM`, applied at render time via `AgentResumeArgv` across the app resume/fork builder, `cmux` CLI surface-restore publisher, and Session Index manual resume.
  - Wrap token-bearing resume/fork commands in `/bin/sh -c '…'` so fish/csh/tcsh parse them; applied only when the token is present and kept outside the cwd guard; cmux-CLI resolves stay unwrapped.
  - Ensure wrapper execution even when `env … claude …` bypasses shell functions and login shells rebuild `PATH`; if the shim var is unset or points to a reaped path, fall back to PATH‑resolved `claude`.
  - Add regression tests for `zsh -lic`, `tcsh`, fish, and stale-shim fallback; bound launcher-shell waits; update launcher-script expectations for long non‑ASCII cwd commands; align kiro tests with the cwd-guard form.

- **Refactors**
  - Move POSIX single-quote helper (`posixSingleQuoted`) to file scope per package policy.

<sup>Written for commit 453c4ad305617d25187a65df7241e602734fca23. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/5721?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Refactor**
  * Claude resume invocations now route through a runtime-resolved wrapper shim and are rendered in a portable shell form so resume/fork commands execute reliably across different shells.
  * Kiro resume commands now include a safe working-directory guard wrapper.

* **Tests**
  * Updated and added regression tests to expect wrapper-based invocations and to exercise wrapper resolution, env/flag preservation, shell compatibility, and edge cases.

### CODE REVIEW

File: cmuxTests/SessionPersistenceTests.swift [hunk 1]
```diff
@@ -1842,12 +1842,138 @@ final class SocketListenerAcceptPolicyTests: XCTestCase {
...
+        process.arguments = ["-lic", resumeCommand]
+        var environment = ["HOME": userHome.path, "ZDOTDIR": userHome.path]
+        // CMUX_CLAUDE_WRAPPER_SHIM is a managed terminal env var inherited by the -lic shell.
+        environment["CMUX_CLAUDE_WRAPPER_SHIM"] = shimURL.path
+        process.environment = environment
+        process.standardInput = FileHandle.nullDevice
+        process.standardOutput = FileHandle.nullDevice
+        process.standardError = FileHandle.nullDevice
+        try process.run()
+        process.waitUntilExit()
```
[the remark this answers is not in the document]
Comment (austinywang) [reply cmuxTests/SessionPersistenceTests.swift]: Fixed in 7e692af23 — the shell waits now use a termination-handler semaphore with a 30s deadline and fail with a clear message instead of hanging (applied to the zsh test and the shared tcsh/fish launcher-test helper).

*— Claude Code*

File: cmuxTests/SessionPersistenceTests.swift [hunk 2]
```diff
@@ -3233,7 +3683,7 @@ final class SocketListenerAcceptPolicyTests: XCTestCase {
 
         XCTAssertEqual(
             snapshot.resumeCommand,
-            "'env' 'NODE_OPTIONS=--trace-warnings' 'claude' '--resume' 'claude-session-node-options' '--model' 'sonnet'"
+            "/bin/sh -c " + shellQuotedForTest("'env' 'NODE_OPTIONS=--trace-warnings' \"$([ -x \"${CMUX_CLAUDE_WRAPPER_SHIM:-}\" ] && printf '%s' \"$CMUX_CLAUDE_WRAPPER_SHIM\" || printf claude)\" '--resume' 'claude-session-node-options' '--model' 'sonnet'")
```
[the remark this answers is not in the document]
Comment (austinywang) [reply cmuxTests/SessionPersistenceTests.swift:3686]: Not an issue — the stripped size flag is cmux's own injected pair, not user tuning. AgentLaunchEnvironmentPolicy drops the heap cap only when it is exactly the cmux-injected --max-old-space-size=4096 (inline or space-separated) AND the cmux restore-node-options.cjs require marked the captured NODE_OPTIONS as cmux-written; a user's own value (different size, or 4096 without the restore module — asserted preserved at line ~3633) survives, as do user flags like --trace-warnings. The cmux wrapper re-applies its NODE_OPTIONS on the resumed launch. This logic predates this PR (5b31ba2f8); the diff only re-wrapped the expectation strings around it.

*— Claude Code*
